### PR TITLE
Actually check for approvals

### PR DESCRIPTION
**Pull request by genevieveluyt**
As it turns out the commit status returned by Github does not account for required approvals (if the branch requires approvals but the PR is merged without them, the commit "status" is still "SUCCESS"). This checks that the PR actually meets the approvals requirements by comparing the number of PR reviews that have state "APPROVED" to the branch protection requirements for the base branch of the PR.

### CODE REVIEW

File: repository/github/queries.go [hunk 1]
```diff
@@ -77,15 +77,89 @@ func newCommitInfoResult(ctx context.Context, ghc ghGraphQLClient, commitURL str
 		return repository.Commit{}, err
 	}
 	queryResult := new(commitInfoQuery)
+
```
Comment (catherinejones) [inline repository/github/queries.go]: This file is terrifying to review, could you split it into multiple files based on query? 😅

File: checks/approved/check.go [hunk 2]
```diff
@@ -75,14 +90,17 @@ func isSigned(commit repository.Commit) bool {
 	return commit.IsSigned
 }
 
-// isMergeCommit checks that the commit is a merge commit
-func isMergeCommit(commit repository.Commit) bool {
+// isApprovedMergeCommit checks that the commit is a merge commit
+func isApprovedMergeCommit(commit repository.Commit, branch repository.Branch) (passed bool, reason error) {
```
Comment (catherinejones) [inline checks/approved/check.go]: Where is `branch` being used here?

Comment (genevieveluyt) [reply checks/approved/check.go]: Oops missed this when resolving conflicts, good catch!

**Comment by genevieveluyt** [comment]
Re `createHandler`, I didn't write that function but I don't see any reason to keep the unused parameter...

**Pull request merged**